x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/analysis/AnalyzerTests.java

```diff
@@ -2519,37 +2536,49 @@ public void testDenseVectorEvalCastingSimilarityFunctions() {
         if (EsqlCapabilities.Cap.COSINE_VECTOR_SIMILARITY_FUNCTION.isEnabled()) {
             checkDenseVectorEvalCastingSimilarityFunction("v_cosine(float_vector, query)");
             checkDenseVectorEvalCastingSimilarityFunction("v_cosine(byte_vector, query)");
+            if (EsqlCapabilities.Cap.GENERIC_VECTOR_FORMAT.isEnabled()) {
+                checkDenseVectorEvalCastingSimilarityFunction("v_cosine(bfloat16_vector, query)");
+            }
         }
         if (EsqlCapabilities.Cap.DOT_PRODUCT_VECTOR_SIMILARITY_FUNCTION.isEnabled()) {
             checkDenseVectorEvalCastingSimilarityFunction("v_dot_product(float_vector, query)");
             checkDenseVectorEvalCastingSimilarityFunction("v_dot_product(byte_vector, query)");
+            if (EsqlCapabilities.Cap.GENERIC_VECTOR_FORMAT.isEnabled()) {
+                checkDenseVectorEvalCastingSimilarityFunction("v_dot_product(bfloat16_vector, query)");
+            }
         }
         if (EsqlCapabilities.Cap.L1_NORM_VECTOR_SIMILARITY_FUNCTION.isEnabled()) {
             checkDenseVectorEvalCastingSimilarityFunction("v_l1_norm(float_vector, query)");
             checkDenseVectorEvalCastingSimilarityFunction("v_l1_norm(byte_vector, query)");
+            if (EsqlCapabilities.Cap.GENERIC_VECTOR_FORMAT.isEnabled()) {
```

Comment:
we don't need to wrap this in an if statement - the capability is always enabled in snapshot and release builds.
for the CSV tests/multi cluster tests the cap checks make sense, because we also run these tests in bwc mode.
also it would be useful to merge main here since with https://github.com/elastic/elasticsearch/pull/139365 we took the vector similarity functions to tech preview and we removed the capability checks in AnalyzerTests